ethcore/sync/src/chain/mod.rs

```diff
@@ -672,6 +700,8 @@ pub struct ChainSync {
 	/// Connected peers pending Status message.
 	/// Value is request timestamp.
 	handshaking_peers: HashMap<PeerId, Instant>,
+	/// Requests, that can not be processed at the moment
+	delayed_requests: Vec<(PeerId, u8, Vec<u8>)>,
```

Comment:
I am a bit worried that this is unbounded. How do we know it doesn't grow too big?

Reply:
I addressed this by cleaning requests for disconnected peer (in terms of the other your question)